util/cmake/SeqAnBuildSystem.cmake

```diff
@@ -247,14 +254,25 @@ macro (seqan_build_system_init)
         if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
             set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -ipo -no-prec-div -fp-model fast=2 -xHOST")
         endif ()
-    elseif (SEQAN_ARCH_SSE4)
-        message (STATUS "Building optimized binaries up to SSE4 and POPCNT.")
-        set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -mmmx -msse -msse2 -msse3 -mssse3 -msse4")
-        if (NOT ${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
-            set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -mpopcnt")
-        endif ()
-        if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
-            set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -ipo -no-prec-div -fp-model fast=2")
+    else ()
+        if (SEQAN_ARCH_SSE4)
+            if (SEQAN_ARCH_AVX2)
+                message (STATUS "Building optimized binaries up to AVX2 and POPCNT.")
+                set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -mavx -mavx2")
+            else ()
+                message (STATUS "Building optimized binaries up to SSE4 and POPCNT.")
+            endif ()
+            set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -mmmx -msse -msse2 -msse3 -mssse3 -msse4")
+            if (NOT ${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
+                set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -mpopcnt")
+            endif ()
+            if (CMAKE_CXX_COMPILER_ID MATCHES "Intel")
+                set (SEQAN_CXX_FLAGS "${SEQAN_CXX_FLAGS} -ipo -no-prec-div -fp-model fast=2")
+            endif ()
+        else ()
+            if (NOT CMAKE_CXX_COMPILER_ID MATCHES "MSVC")  # Explicitly disable extended vector instruction sets.
```

Comment:
I don't think we need this anymore, now that the simd_module is fixed. I have read up on the issue, and `-msse3` is set on osx by clang and gcc, but none of the other flags. On all other platforms, none of the flags are set anyway.